pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -348,17 +350,17 @@ func (m *ManagerImpl) Start(activePods ActivePodsFunc, sourcesReady config.Sourc
 	// Loads in allocatedDevices information from disk.
 	err := m.readCheckpoint()
 	if err != nil {
-		klog.ErrorS(err, "Continue after failing to read checkpoint file. Device allocation info may NOT be up-to-date")
+		logger.Error(err, "Continue after failing to read checkpoint file. Device allocation info may NOT be up-to-date")
 	}
 
-	return m.server.Start()
+	return m.server.Start(ctx)
```

Comment:
this seems wrong. gRPC server should take another context as input - otherwise in future initialization logic of kubelet may cancel the long running gRPC server

Reply:
Thanks for the quick review. How about adding a new ctx field into the [server struct](https://github.com/kubernetes/kubernetes/blob/befc91aec12e7e4ab12f321ce82cf189f7fd0f12/pkg/kubelet/cm/devicemanager/plugin/v1beta1/server.go#L50), similar to how [backgroundCtx](https://github.com/kubernetes/kubernetes/blob/befc91aec12e7e4ab12f321ce82cf189f7fd0f12/pkg/kubelet/cm/dra/plugin/dra_plugin_manager.go#L52) is used in DRAPluginManager?

Reply:
Thanks for clarifying